src/test/common/test_split.cc

```diff
@@ -0,0 +1,79 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+
+#include "common/split.h"
+#include <algorithm>
+#include <gtest/gtest.h>
+
+namespace ceph {
+
+TEST(split, iterator)
+{
+  const auto parts = split("a,b");
+  auto i = parts.begin();
+  ASSERT_NE(i, parts.end());
+  ASSERT_EQ(2, std::distance(i, parts.end()));
+
+  // copy the iterator to test LegacyForwardIterator's multipass guarantee
+  auto j = i;
+  ASSERT_EQ(i, j);
+  ASSERT_NE(j, parts.end());
+
+  ASSERT_EQ("a", *i);
+  EXPECT_EQ(1, i->size()); // test operator->
+  EXPECT_EQ("a", *i++); // test post-increment of i
+  ASSERT_NE(i, parts.end());
+  EXPECT_EQ("b", *i);
+  EXPECT_EQ(std::next(i), parts.end());
+
+  ASSERT_EQ("a", *j); // test that i++ left j unmodified
+  EXPECT_EQ("b", *++j); // test pre-increment of j
+  EXPECT_EQ("b", *j);
+  EXPECT_EQ(std::next(j), parts.end());
+  EXPECT_EQ(i, j);
+
+  // test comparions against default-constructed 'singular iterator'
+  split::iterator k;
+  EXPECT_EQ(k, parts.end());
+  EXPECT_NE(k, i);
+  EXPECT_NE(k, j);
+  EXPECT_NE(k, parts.begin());
+}
+
+using string_list = std::initializer_list<std::string_view>;
+
+bool operator==(const split& lhs, const string_list& rhs) {
+  return std::equal(lhs.begin(), lhs.end(), rhs.begin(), rhs.end());
+}
+bool operator==(const string_list& lhs, const split& rhs) {
+  return std::equal(lhs.begin(), lhs.end(), rhs.begin(), rhs.end());
+}
+
+TEST(split, split)
+{
+  EXPECT_EQ(string_list({}), split(""));
+  EXPECT_EQ(string_list({}), split(","));
```

Comment:
String splitting in most languages would return two empty strings here.

Reply:
The interface we're replacing merges runs of delimiters.